### DISCUSSION TITLE
Organisation switcher

### DISCUSSION THREAD

**Discussion by dombillington** [ministryofjustice/moj-frontend#239] [RELEASED: Components, pages and patterns]
## Existing component

- [Organisation switcher in MoJ Pattern Library](https://design-patterns.service.justice.gov.uk/components/organisation-switcher/)

The existing organisation switcher pattern in the MOJ library floats the change link to the right of the layout. At widescreen desktop resolutions, the relationship between these elements isn’t clear. 61% of users in DPS have a desktop resolution of 1920 x 1080.

### What we've changed

In DPS, we’ve iterated on this layout to move the change link next to the location. This makes it clearer to users that the change link is associated with the location displayed.

The link text in DPS is: ‘change your location'. We’ve included the pronoun to make it clear to the user they’re changing their location, not the location of a prisoner.

### Service: Digital Prison Services (DPS)

![Screenshot 2021-07-23 at 09 52 21](https://user-images.githubusercontent.com/19429717/126766401-7c60cfdd-ef90-4093-b8be-78a33813bdf8.png)

**Comment by Whersi** [comment]
## Ticket number

https://github.com/ministryofjustice/moj-frontend/issues/284

The details of the ticket is as follows:

## Summary

The [Organisation switcher](https://design-patterns.service.justice.gov.uk/components/organisation-switcher/) currently floats the link to the right, which can be an issue for accessibility: the link can be hard to see and click, and isn't clearly associated with the organisation name.

A more appropriate design (and one already adopted in some services) would be to button the link alongside the text:
![image](https://user-images.githubusercontent.com/100852/146202309-13200da9-e4bd-4788-9267-47405c7c8b00.png)

Since this may be a visually-breaking-change for some services, we should make this new design opt-in as a minor release, and then change the default behaviour in a future release. Therefore I propose adding a new option to the Nunjucks macro: `link.position` can be "inline" or "right" (currently defaulting to "right").

The "inline" option could use standard GOV.UK Design System classes rather than needing bespoke MOJ Frontend classes. For example:
```html
<div class="moj-organisation-nav" aria-label="Organisation switcher">
  <strong class="govuk-body govuk-!-font-weight-bold">Bristol (HMP)</strong>
  <a href="#" class="govuk-link">Change your location</a>
</div>
```

## Describe alternatives you've considered

Due to the simplicity of this component, I'm not sure there are any other suitable options that don't amount to a full redesign.

## Additional context

Raised in the accessibility audit and later [via Slack](https://mojdt.slack.com/archives/CH5RUSB27/p1639576556046300).
